hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/MappingModelCreationHelper.java

```diff
@@ -762,7 +764,7 @@ private static void interpretPluralAttributeMappingKeyDescriptor(
 
 			final BasicValuedModelPart simpleFkTargetPart = (BasicValuedModelPart) fkTargetPart;
 
-			final String keyTableExpression = getTableIdentifierExpression( bootValueMappingKey.getTable(), creationProcess );
+			final String keyTableExpression = collectionTableName;//getTableIdentifierExpression( bootValueMappingKey.getTable(), creationProcess );
```

Comment:
This is for changing the `containingTableExpression` from e.g. `AbstractTablePerClassEntity` to `(select ... from table1 union all select ... from table2 ...)`, to match what the entity persister reports as its table name. Needed for some assert statements.